pkg/provisioner/ironic/raid.go

```diff
@@ -43,6 +43,13 @@ func setTargetRAIDCfg(p *ironicProvisioner, raidInterface string, ironicNode *no
 		p.log.Info("rootDeviceHints is used, the first volume of raid will not be set to root")
 	}
 
+	updater := updateOptsBuilder(p.debugLog)
+	updater.SetTopLevelOpt("raid_interface", targetRaidInterface, ironicNode.RAIDInterface)
```

Comment:
> This is never the case, as software RAID is not driver-specific.

In this case, why do we need to update the `raid_interface` here? If we want to configure software RAID, this will change the node's `raid_interface` to `agent`. If the node had hardware RAID before this would cause the problem that @zaneb  pointed out below.

> If the host was configured with hardware RAID and we're removing it, we use the hardware RAID driver to do so, as you'd expect. But if we're simultaneously replacing the hardware RAID config with a software RAID config, then we'll try to use the software RAID driver to remove the hardware RAID config, which is surprising

Reply:
> In this case, why do we need to update the raid_interface here? If we want to configure software RAID, this will change the node's raid_interface to agent.

Yes. Otherwise you will not be able to build software RAID.